scripts/irods/test/test_symlink_operations.py

```diff
@@ -246,24 +246,28 @@ def test_irsync_iput_recursive_dangling_symlink_issues_3072_3988(self):
             # Iterate through each of the tests:
             ########
             for teststring in test_list:
+                with self.subTest(teststring):
+                    ignore_symlinks = '--link' in teststring
 
-                cmd = teststring.format(**locals())
-                stdout,stderr,_ = self.user0.run_icommand(cmd)
+                    cmd = teststring.format(**locals())
+                    stdout,stderr,_ = self.user0.run_icommand(cmd)
 
-                self.assertIn(recurse_fail_string, stdout,
-                              '{0}: Expected stdout: "...{1}...", got: "{2}"'.format(cmd, recurse_fail_string, stdout))
+                    if ignore_symlinks:
+                        self.assertEqual(str(), stdout, '{}: Expected stdout: "", got: "{}"'.format(cmd, stdout))
+                        self.assertEqual(str(), stderr, '{}: Expected stderr: "", got: "{}"'.format(cmd, stderr))
 
-                self.assertIn('ERROR: USER_INPUT_PATH_ERR: No such file or directory', stderr,
-                              '{0}: Expected stderr: "...{1}...", got: "{2}"'.format(cmd,
-                                    'ERROR: USER_INPUT_PATH_ERR: No such file or directory', stderr)
-                )
+                    else:
+                        self.assertIn(recurse_fail_string, stdout,
+                                      '{0}: Expected stdout: "...{1}...", got: "{2}"'.format(cmd, recurse_fail_string, stdout))
 
-                # Make sure the single valid file under dir1
-                # was not moved into the vault
-                self.user0.assert_icommand('ils -l {target_collection_path}/newname'.format(**locals()),
-                            'STDERR_SINGLELINE',
-                        '{target_collection_path}/newname does not exist or user lacks access permission'.format(**locals())
-                )
```

Comment:
If you're wondering why this assertion was changed, please see the following history...

The test was introduced in this commit and originally asserted that the dangling symlink does not appear in iRODS:
https://github.com/irods/irods/commit/bb4cb508e822794e3aec8eb43140cea3c829f96c

The test was changed in this commit to assert that the valid file would not be uploaded:
https://github.com/irods/irods/commit/e37c529f2beabc66cbda529bd2e06f55f5537008

I'm not really sure how this was passing before because from my changes I do not see how this is preventing the "pre-flight" scan from checking for the dangling symlink. What we have now is *partial success* because the dangling symlink just fails to upload later. We can discuss whether this is the expected behavior and I can look into threading that particular needle.

Reply:
After some discussion we have determined that this assertion should remain for the `ignore_symlinks == False` case. Update to follow...

Reply:
As it turns out, this is not a problem with the new changes but with the test. In the past, this test never uploaded anything due to the incorrect handling of `--link` and so never needed to clean up after itself between sub-tests. Now that `--link` is being handled correctly, tests which use `--link` will upload things and the test needs to clean up the test collection between runs.

Reply:
Done